extensions/pkg/webhook/webhook.go

```diff
@@ -59,24 +64,34 @@ type Args struct {
 	Provider   string
 	Name       string
 	Path       string
+	Target     string
 	Predicates []predicate.Predicate
 	Validators map[Validator][]Type
 	Mutators   map[Mutator][]Type
 }
 
 // New creates a new Webhook with the given args.
 func New(mgr manager.Manager, args Args) (*Webhook, error) {
-	logger := log.Log.WithName(args.Name).WithValues("provider", args.Provider)
+	var (
+		objTypes   []Type
+		actionType string
 
-	// Create handler
-	builder := NewBuilder(mgr, logger)
+		logger  = log.Log.WithName(args.Name).WithValues("provider", args.Provider)
+		builder = NewBuilder(mgr, logger)
+	)
 
 	for val, objs := range args.Validators {
 		builder.WithValidator(val, objs...)
+		objTypes = append(objTypes, objs...)
+		if actionType != ActionMutating {
+			actionType = ActionValidating
+		}
 	}
 
 	for mut, objs := range args.Mutators {
 		builder.WithMutator(mut, objs...)
+		objTypes = append(objTypes, objs...)
+		actionType = ActionMutating
```

Comment:
`actionType` would always be set to `ActionMutating` if the length of `args.Mutator` > 0 regardless of the length of  `args.Validator`. Is this expected?

Reply:
Yes this is expected. Theoretically, you can mix mutators and validators via the builder but the result will still be a single webhook. As soon as one mutator is involved you need a `MutatingwebhookConfiguration`.

Reply:
Should we rather add some validation in the beginning of this function to prevent mixing validators and mutators? It's a bit confusing overall, and one shouldn't do this, or is there a scenario where you would?